samples: openthread: add GPIO diag configuration for nRF52840 dongle (#11352)
nrfconnect/sdk-nrf: samples/openthread/cli/README.rst

.. note::
    If you disable the :kconfig:option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER` feature set, you can enable the diagnostic module with the :kconfig:option:`CONFIG_OPENTHREAD_DIAG` Kconfig option.

Additionally for the ``nrf52840dongle_nrf52840`` build target, diagnostic GPIO commands have been enabled by ``openthread_config`` node in :file:`boards/nrf52840dongle_nrf52840.overlay`.
This node has been configured for P0.19 pin, which is connected to reset on nRF52840 Dongle. By setting it in output mode with low state reset is pulled to GND, which results in device rebooting without skipping the bootloader.

To trigger rebooting to bootloader run below commands on the device:

.. code-block:: console

   uart:~$ ot diag start
   start diagnostics mode
   status 0x00
   Done

>    uart:~$ ot diag gpio mode 0 out

Reviewer: Isn't `Done` printed after this?

Author: It isn't as device immediately closes the connection because of triggering the reset

.. note::
    If you disable the :kconfig:option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER` feature set, you can enable the diagnostic module with the :kconfig:option:`CONFIG_OPENTHREAD_DIAG` Kconfig option.

> Additionally for the ``nrf52840dongle_nrf52840`` build target, diagnostic GPIO commands have been enabled by ``openthread_config`` node in :file:`boards/nrf52840dongle_nrf52840.overlay`.
> This node has been configured for P0.19 pin, which is connected to reset on nRF52840 Dongle.

Reviewer 2: ```suggestion
For the ``nrf52840dongle_nrf52840`` build target, diagnostic GPIO commands can also be used.
They are enabled in the ``openthread_config`` node in the :file:`boards/nrf52840dongle_nrf52840.overlay` file, with the node configured for the **P0.19** pin, which is connected to reset on the nRF52840 Dongle.
```
Should "reset" be "the **RESET** button" here?

Author: Maybe "the RESET pin" or "the RESET signal"? Button itself is connected to RESET path so it doesn't sound intuitive to me
![image](https://github.com/nrfconnect/sdk-nrf/assets/115064000/34334225-fc44-4629-88e3-1b2d8b0b54bd)

Reviewer 2: Yeah, that looks like "the **RESET** pin" would be correct. I did take a quick look at the diagram, but didn't notice it there.

> They are enabled in the ``openthread_config`` node in the :file:`boards/nrf52840dongle_nrf52840.overlay` file, with the node configured for the **P0.19** pin, which is connected to the RESET pin on the nRF52840 Dongle.

Reviewer 2: ```suggestion
They are enabled in the ``openthread_config`` node in the :file:`boards/nrf52840dongle_nrf52840.overlay` file, with the node configured for the **P0.19** pin, which is connected to the **RESET** pin on the nRF52840 Dongle.
```

> By setting it in output mode with low state reset is pulled to GND, which results in device rebooting without skipping the bootloader.

Reviewer 2: ```suggestion
The pin is set in output mode with a low state  so that reset is pulled to **GND**, which results in the device rebooting without skipping the bootloader.
```
Same as above, should "reset" be "the **RESET** button"?

> The pin is set in output mode with a low state  so that the RESET pin is pulled to **GND**, which results in the device rebooting without skipping the bootloader.

Reviewer 2: ```suggestion
The pin is set in output mode with a low state  so that the **RESET** pin is pulled to **GND**, which results in the device rebooting without skipping the bootloader.
```

> The pin is set in output mode with a low state  so that the **RESET** pin is pulled to **GND**, which results in the device rebooting without skipping the bootloader.

Reviewer 2: ```suggestion
The pin is set in output mode with a low state so that the **RESET** pin is pulled to **GND**, which results in the device rebooting without skipping the bootloader.
```
Not a blocker, but if you happen to make other changes, there's an extra space here.

Author: Fixed. Thanks for catching that :)

This functionality is not enabled by other commands like ``factoryreset`` as they can only trigger software reset, skipping the bootloader.

> To trigger rebooting to bootloader run below commands on the device:

Reviewer 2: ```suggestion
To reboot to the bootloader, run the following commands on the device:
```
